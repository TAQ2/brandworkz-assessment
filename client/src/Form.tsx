import React from "react";
import { GoSearch } from "react-icons/go";
import styled from "styled-components";

import { colours, screenBreakpoints } from "./theme";

export const Input = styled.input`
  width: 400px;
  padding: 0.5em;
  font-weight: bold;
  font-size: 1em;
  border: none;

  @media (max-width: ${screenBreakpoints.tablet}px) {
    width: 300px;
  }

  @media (max-width: ${screenBreakpoints.small}px) {
    width: 225px;
  }
`;

const Form: React.FC<{
  handleUserInput: (e: React.FormEvent<HTMLInputElement>) => void;
  userInput: string;
  handleSubmit: () => void;
}> = ({ handleUserInput, userInput, handleSubmit }) => {
  return (
    <div
      style={{
        marginBottom: "3em",
        display: "inline-block",
        border: `1px solid ${colours.primary}`
      }}
    >
      <Input
        onChange={handleUserInput}
        value={userInput}
        placeholder="Type a city"
      />
      <button
        onClick={handleSubmit}
        style={{
          fontSize: "1em",
          padding: "0.5em",
          border: "none",
          cursor: "pointer",
          outline: "0"
        }}
      >
        <GoSearch />
      </button>
    </div>
  );
};

export default Form;
