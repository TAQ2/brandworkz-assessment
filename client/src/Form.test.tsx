import React from "react";
import { shallow, mount } from "enzyme";

import Form from "./Form";

it("renders without crashing", () => {
  const formProps = {
    handleSubmit: () => {},
    userInput: "Test",
    handleUserInput: () => {}
  };
  const wrapper = shallow(<Form {...formProps} />);
  expect(wrapper).toMatchSnapshot();
});

it("displays the user input in an <input /> and updates accordingly", () => {
  const formProps = {
    handleSubmit: () => {},
    userInput: "Test",
    handleUserInput: () => {}
  };
  const wrapper = mount(<Form {...formProps} />);
  expect(wrapper.find("input").instance().value).toEqual(formProps.userInput);

  // now update the userInput prop
  const newUserInput = "Different Test";
  wrapper.setProps({ ...formProps, userInput: newUserInput });
  expect(wrapper.find("input").instance().value).toEqual(newUserInput);
});

it("handleUserInput is called when the input onChange is called", () => {
  const spy = jest.fn();
  const formProps = {
    handleSubmit: () => {},
    userInput: "Test",
    handleUserInput: spy
  };
  const wrapper = mount(<Form {...formProps} />);

  const event = { target: { value: "event" } };
  wrapper.find("input").simulate("change", event);

  expect(spy).toBeCalledTimes(1);
  expect(spy).toBeCalledWith(expect.objectContaining(event));
});

it("handleSubmit is called when the button is clicked", () => {
  const spy = jest.fn();
  const formProps = {
    handleSubmit: spy,
    userInput: "Test",
    handleUserInput: () => {}
  };
  const wrapper = mount(<Form {...formProps} />);

  wrapper.find("button").simulate("click");
  expect(spy).toBeCalledTimes(1);
});
