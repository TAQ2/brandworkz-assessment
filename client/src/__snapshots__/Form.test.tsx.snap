// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  style={
    Object {
      "border": "1px solid purple",
      "display": "inline-block",
      "marginBottom": "3em",
    }
  }
>
  <styled.input
    onChange={[Function]}
    placeholder="Type a city"
    value="Test"
  />
  <button
    onClick={[Function]}
    style={
      Object {
        "border": "none",
        "cursor": "pointer",
        "fontSize": "1em",
        "outline": "0",
        "padding": "0.5em",
      }
    }
  >
    <GoSearch />
  </button>
</div>
`;
